src/common/btree.c

```diff
@@ -597,7 +597,7 @@ static inline int
 btr_node_size(struct btr_context *tcx)
 {
 	return sizeof(struct btr_node) +
-		tcx->tc_tins.ti_root->tr_node_size * btr_rec_size(tcx);
+		(int)tcx->tc_tins.ti_root->tr_node_size * btr_rec_size(tcx);
```

Comment:
This won't cause overflow because the uint8_t will be casted to 64bits before doing math.